server/controllers/expenses.js

```diff
@@ -182,9 +180,9 @@ module.exports = (app) => {
           }
         }
       } else {
-        return preapprovalDetails => {
-          const maxAmount = Number(preapprovalDetails.maxTotalAmountOfAllPayments);
-          const currency = preapprovalDetails.currencyCode;
+        return details => {
+          const maxAmount = Number(details.maxTotalAmountOfAllPayments) - Number(details.curPaymentsAmount);
```

Comment:
I think original code was incorrect, right? `checkIfEnoughFunds` was always taking the max amount on the key, even if key had already been partly used.